### PR TITLE
internal/db: do not run TimescaleDB migrations against singleton database

### PR DESCRIPTION
**Pull request by slimsag**
In our DB testing code we make an assumption that we can run all database migrations against a single testing database. This holds true for frontend and codeintel migrations, because we intentionally designed codeintel migrations to be ran in the same DB for testing/dev purposes only - but this does not hold true for Code Insights which runs a separate TimescaleDB instance (it is easier to run it separately in Docker than install it as a Postgres extension in all of our dev/testing environments.)

This change merely makes it so we don't run TimescaleDB migrations against our singleton testing/dev Postgres DB.

Signed-off-by: Stephen Gutekanst <stephen@sourcegraph.com>

### CODE REVIEW

File: internal/db/dbconn/migration.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 // the raw migration assets to run to migrate the target schema to a new version.
 var databases = map[string]struct {
 	MigrationsTable string
+	TimescaleDB     bool
```
Comment (efritz) [inline internal/db/dbconn/migration.go:25]: Seems like this would be better inclusive (`TargetsPostgres`) rather than exclusive (`TimescaleDB`).

**Review by efritz — COMMENTED** [review]
Would it be a better solution to separate postgres from the code intel database rather than making exclusions to new things (that are probably more probably likely to act like code insights database than a code intelligence database)?

File: internal/db/dbconn/migration.go (hunk 1 above)
Comment (slimsag) [reply internal/db/dbconn/migration.go:25]: Definitely

**Comment by slimsag** [comment]
@efritz Im not sure I understand what you mean exactly, can you elaborate?

**Comment by efritz** [comment]
Should we also separate Postgres and codeintel-db here? Instead of making timescale the odd one out, we can make the main "Postgres" db less of a "main" store and all three dbs would have equal tooling.

**Comment by slimsag** [comment]
Ahh, got it. I do think that is something we should do, I'll take a pass at that.

**Comment by slimsag** [comment]
@efritz PTAL, I think this is likely a better step in the right direction?

File: internal/database/dbstore_db_test.go [hunk 2]
```diff
@@ -17,23 +17,27 @@ func TestMigrations(t *testing.T) {
...
+			t.Errorf("error running initial migrations: %s", err)
 		}
 	}
 
-	for _, databaseName := range dbconn.DatabaseNames {
-		t.Run(databaseName, func(t *testing.T) {
+	for _, database := range []*dbconn.Database{
+		dbconn.Frontend,
+		dbconn.CodeIntel,
+	} {
```
Comment (slimsag) [inline internal/database/dbstore_db_test.go:31]: Note: Still using a few lists like this in some places to use "the same DB for frontend and codeintel" but I will start fixing that in subsequent PRs where I aim to add code-insights-equivilents to these locations.

For now, it at least makes this "we use multiple DBs here which is kinda clunky" clear & explicit

**Review by efritz — APPROVED** [review]
Ain't no confusion here!

File: internal/database/dbstore_db_test.go (hunk 2 above)
Comment (efritz) [reply internal/database/dbstore_db_test.go:31]: Yeah I think this is worlds better than tacking a third one on and doing retroactive edge case fixes.

**Comment by efritz** [comment]
Thanks for the effort here, @slimsag! Separating the codeintel db and the frontend db is a very very low priority at this point as it only affects dev and we have checks on startup to ensure it doesn't actually happen.

This PR really minimizes additional muck so I think we can 🚀 with this guy now.

**Review by asdine — APPROVED** [review]

**Pull request merged**